alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunner.kt

```diff
@@ -292,7 +301,25 @@ class MonitorRunner(
                         val searchResponse: SearchResponse = client.suspendUntil { client.search(searchRequest, it) }
                         results += searchResponse.convertToMap()
                     }
-                    else -> {
+                    is HttpInput -> {
+                        val response: HttpResponse = httpClient.client.suspendUntil {
+                            httpClient.client.execute(input.toGetRequest(), it) }
+                        // Check response content length is not larger than 100MB
+                        val contentLengthHeader = response.getFirstHeader("Content-Length").value
+                        val contentLength = if (contentLengthHeader != null) {
+                            logger.info("Content length is $contentLengthHeader")
+                            contentLengthHeader.toInt()
+                        } else {
+                            logger.info("Content-Length header does not exist, check estimate size of content.")
+                            val content = response.entity.content
+                            logger.info("Content has estimate number of ${content.available()} bytes.")
+                            content.available()
+                        }
+                        if (contentLength > httpClient.MAX_CONTENT_LENGTH) {
+                            throw Exception("Response content size: $contentLength, is larger than ${httpClient.MAX_CONTENT_LENGTH}.")
+                        }
+                        results += response.toMap()
```

Comment:
Does this work if the response is not a JSON response (for e.g. plain text)?

Reply:
No, the method I'm using now will throw an IOException if there's unrecognized token. I have a [test](https://github.com/ann3431/alerting/blob/005b5c2e3236f945a5207453ccf31d18bd56044b/alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt#L585) in `MonitorRunnerIT` that uses the _cat/indices API which returns plain text.

Reply:
Why do we require the output to be JSON?  What about XML, CSV, HTML, plain text which are just as common response `Content-Type`s.   If the response is not JSON we should be including it in `results` as a properly escaped strings.

Reply:
We are planning to support more response types such as plain text, Smile, CBOR, YAML. The types you mentioned are also nice to have. But I won't be available to include more types for initial release. I can create an issue for this once this is pushed to master.  Thanks for your suggestion.